src/reporters/json.ts

```diff
@@ -51,7 +65,7 @@ type OutputFields = {
   step?: Partial<Step>;
   error?: Error;
   root_fields?: Record<string, unknown>;
-  payload?: Record<string, unknown>;
+  payload?: Payload | NetworkInfo;
```

Comment:
If we have to type this, proper way would be to type it for each `type` -> `journey/start`, `journey/end` and such. I am not 100% sure if this required. 

One other way is to use the `Partial` type and keep it minimal and add Record<string, unknown> for the rest.

Reply:
i was almost about to write conditional types, but didn't want to spend too much time on this PR :D

Reply:
yeah lets not go down the rabbit hole and spend too much time on this, we will eventually get rid of payload at some point. lets keep it as it is.